docs/modules/ROOT/pages/runbooks/APPUiOCloudReportingDatabaseBackfillingFailed.adoc

```diff
@@ -14,3 +14,48 @@ Deleting the failed job will clear this alert.
 include::partial$runbooks/common_debugging_steps.adoc[]
 +
 * Check Prometheus/Thanos connectivity and logs.
+
+[IMPORTANT]
+After fixing the underlying issue, you need to backfill all gaps in the reporting database.
+
+== icon:terminal[] Backfill gaps in reporting database
+
+[NOTE]
+====
+This runbook expects `jq` to be installed.
+
+On macOS it expects the `date` command from GNU coreutils to be installed as `gdate` (`brew install coreutils`).
+====
+
+. Find failed jobs
+
+[source,bash]
+----
+CRONJOB_NAME=<name of the cronjob that has failing jobs>
+
+query_name=$(kubectl get jobs --selector=cron-job-name=$CRONJOB_NAME -ojson \
+  | jq -r '[.items[] | select((.status.conditions[] | select( .type == "Failed"  ) | length ) >  0)]
+    | [.[].metadata.annotations["query-name"]] | first')
+echo $query_name
```

Comment:
This seems a bit convoluted? You're basically telling the reader to look for the cronjob with failed jobs and then execute a query to find the failed job and get the annotation from it.

I mean it works but wouldn't for example the cronjob itself have the same `query-name` annotation

Reply:
Yeah you're right. I actually just copy pasted them. But this can be simplified.